impeller/aiks/experimental_canvas.cc

```diff
@@ -44,6 +45,96 @@ static void ApplyFramebufferBlend(Entity& entity) {
   entity.SetBlendMode(BlendMode::kSource);
 }
 
+/// End the current render pass, saving the result as a texture, and then
+/// restart it with the backdrop cleared to the previous contents.
+///
+/// This method is used to set up the input for emulated advanced blends and
+/// backdrop filters.
+///
+/// Returns the previous render pass stored as a texture, or nullptr if there
+/// was a validation failure.
+static std::shared_ptr<Texture> FlipBackdrop(
+    std::vector<LazyRenderingConfig>& render_passes,
+    Point global_pass_position,
+    EntityPassClipStack& clip_coverage_stack,
+    ContentContext& renderer) {
+  auto rendering_config = std::move(render_passes.back());
+  render_passes.pop_back();
+
+  // If the very first thing we render in this EntityPass is a subpass that
+  // happens to have a backdrop filter or advanced blend, than that backdrop
+  // filter/blend will end may wind up sampling from the uninitialized texture
+  //
+  // By calling `pass_context.GetRenderPass` here, we force the texture to pass
+  // through at least one RenderPass with the correct clear configuration before
+  // any sampling occurs.
+  //
+  // In cases where there are no contents, we
+  // could instead check the clear color and initialize a 1x2 CPU texture
+  // instead of ending the pass.
+  rendering_config.inline_pass_context->GetRenderPass(0);
+  if (!rendering_config.inline_pass_context->EndPass()) {
+    VALIDATION_LOG
+        << "Failed to end the current render pass in order to read from "
+           "the backdrop texture and apply an advanced blend.";
+    return nullptr;
+  }
+
+  std::shared_ptr<Texture> input_texture =
+      rendering_config.entity_pass_target->Flip(
+          *renderer.GetContext()->GetResourceAllocator());
+
+  if (!input_texture) {
+    VALIDATION_LOG << "Failed to fetch the color texture in order to "
+                      "apply an advanced blend.";
+    return nullptr;
+  }
+
+  render_passes.push_back(LazyRenderingConfig(
```

Comment:
All of the return statements above this level leave the render_passes one entry short before this line pushes a new pass. Should the pop be moved down to this level (but then there is the issue that the render_pass at the top of the stack was `std::move`d out of the stack at the top of the method...

Reply:
hmm, I don't think we should attempt to recover if we hit a validation error - but unlike the entity pass approach we don'thave a good mechanism to bail out of rendering to avoid a later crash.

I think what we'd need here would be some way for the dispatcher to cancel dispatching via returning false, then this method and all of the canvas methods could eventually be made to return false too.

For now, I think I'll balance the render passes with a note.